validator/tick/models.go

```diff
@@ -58,3 +59,62 @@ func contractFeesToProto(contractFees [1024]int64) []int64 {
 	}
 	return protoContractFees
 }
+
+func WasSkippedByArchive(tick uint32, processedTicksIntervalPerEpoch []*protobuff.ProcessedTickIntervalsPerEpoch) (bool, uint32) {
+	if len(processedTicksIntervalPerEpoch) == 0 {
+		return false, 0
+	}
+	for _, epochInterval := range processedTicksIntervalPerEpoch {
+		for _, interval := range epochInterval.Intervals {
+			if tick < interval.InitialProcessedTick {
+				return true, interval.InitialProcessedTick
+			}
+			if tick >= interval.InitialProcessedTick && tick <= interval.LastProcessedTick {
+				return false, 0
+			}
+		}
+	}
+	return false, 0
+}
+
+func GetTickEpoch(tickNumber uint32, intervals []*protobuff.ProcessedTickIntervalsPerEpoch) (uint32, error) {
+	if len(intervals) == 0 {
+		return 0, errors.New("processed tick interval list is empty")
+	}
+
+	for _, epochInterval := range intervals {
+		for _, interval := range epochInterval.Intervals {
+			if tickNumber >= interval.InitialProcessedTick && tickNumber <= interval.LastProcessedTick {
+				return epochInterval.Epoch, nil
+			}
+		}
+	}
+
+	return 0, errors.New(fmt.Sprintf("unable to find the epoch for tick %d", tickNumber))
+}
+
+func GetProcessedTickIntervalsForEpoch(epoch uint32, intervals []*protobuff.ProcessedTickIntervalsPerEpoch) (*protobuff.ProcessedTickIntervalsPerEpoch, error) {
+	for _, interval := range intervals {
+		if interval.Epoch != epoch {
+			continue
+		}
+		return interval, nil
+	}
+
+	return nil, errors.New(fmt.Sprintf("unable to find processed tick intervals for epoch %d", epoch))
+}
+
+func IsLast(tickNumber uint32, epoch uint32, intervals []*protobuff.ProcessedTickIntervalsPerEpoch) (bool, int, error) {
```

Comment:
this function is not self explanatory from the definition, you either redefine it or if you can't change it to make it more explanatory, then adding a comment is mandatory